esp/bindings/http/platform/httpprot.cpp

```diff
@@ -515,13 +495,16 @@ void CHttpThread::returnSocket()
  **************************************************************************/
 void CPooledHttpThread::init(void *param)
 {
-    m_socket.setown((ISocket*)(((void**)param)[0]));
-    m_apport = (CEspApplicationPort*)(((void**)param)[1]);
-    m_MaxRequestEntityLength = *(int*)(((void**)param)[2]);
-    m_is_ssl = *(bool*)(((void**)param)[3]);
-    m_ssctx = (ISecureSocketContext*)(((void**)param)[4]);
-    m_persistentHandler = (IPersistentHandler*)(((void**)param)[5]);
-    m_shouldClose = *(bool*)(((void**)param)[6]);
+    if (!param)
+        throw makeStringException(-1, "CPooledHttpThread::init: invalid param");
+    PooledThreadInfo* ptip = (PooledThreadInfo*)param;
+    m_socket.set(&ptip->socket);
+    m_apport = &ptip->apport;
```

Comment:
I'm guessing you left `m_apport` as a pointer to avoid touching a lot more code? Or is there positive benefit for checking an uninitialized thread? It seems like it would be clearer to change it to a reference, but I understand that is probably very low priority and not something to tackle in this PR.